handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -844,11 +831,12 @@ private void wrap(ChannelHandlerContext ctx, boolean inUnwrap) throws SSLExcepti
                 if (result.getStatus() == Status.CLOSED) {
                     buf.release();
                     buf = null;
-                    promise.tryFailure(SSLENGINE_CLOSED);
+                    SSLException exception = new SSLException("SSLEngine closed already");
```

Comment:
@njhill huh? This is only used within this method and not shared, not sure what issue you see here.

Reply:
@normanmaurer sorry I missed this before somehow. I was referring to the fact that the same exception is propagated to more than one place (e.g. used to complete different futures), so could in theory be rethrown in different threads at the same time. But I think it's a non-issue since it must be the thrower's responsibility to ensure that the thrown exception is not shared, even if it wraps one which is (i.e. a `new ExecutionException` is used on every call to `Future.get()` in both netty and JDK impls).

Incidentally I just noticed there _is_ such an issue w.r.t. how `CancellationException`s are thrown from netty's `AbstractFuture`. It's an obscure/unlikely scenario but multiple threads could be blocked on `get()` for the same future which is subsequently cancelled. The same `CancellationException` will propagate in both threads concurrently which could each independently call `addSuppressed(...)` on it. I can open a PR if you consider this a "bug" :)